### PR TITLE
1040 Fix error handler on lambda async invocation + Disable SQS retries

### PR DESCRIPTION
**Pull request by leite08**
Ticket: 

### Dependencies

- Upstream: https://github.com/metriport/metriport/pull/2326
- Downstream: none

### Description

[context](https://metriport.slack.com/archives/C04T5Q9JHFX/p1719331694970869?thread_ts=1719331458.061259&cid=C04T5Q9JHFX):
- get result lambda payload checks for 2xx instead of 200
- `defaultLambdaInvocationResponseHandler` has the correct lambda name

Just avoiding hitting the OSS API too much if its already failing
- response from outbound PD IHE GW lambda to try max 5 attempts to OSS API

[context](https://metriport.slack.com/archives/C04GEQ1GH9D/p1719347777789909?thread_ts=1719338701.693029&cid=C04GEQ1GH9D):
- disable retries on FHIR converter and server lambdas

### Testing

upstream PR
 
### Release Plan

- [ ] Merge this

### CODE REVIEW

File: packages/core/src/external/aws/lambda.ts [hunk 1]
```diff
@@ -97,9 +100,12 @@ export function getLambdaResultPayload({
   failOnEmptyResponse?: boolean;
   log?: typeof console.log;
 }): string | undefined {
-  if (result.StatusCode !== 200) {
+  if (!result.StatusCode || result.StatusCode < 200 || result.StatusCode > 299) {
```
Comment (leite08) [inline packages/core/src/external/aws/lambda.ts:103]: culprit of the [issue at staging](https://metriport.slack.com/archives/C04T5Q9JHFX/p1719331694970869?thread_ts=1719331458.061259&cid=C04T5Q9JHFX), it was invoking OK, just the response code was 202

File: packages/core/src/external/carequality/ihe-gateway-v2/ihe-gateway-v2-logic.ts [hunk 2]
```diff
@@ -157,6 +157,8 @@ export async function createSignSendProcessXCPDRequest({
     try {
       // TODO not sure if we should retry on timeout
       await executeWithNetworkRetries(async () => axios.post(pdResponseUrl, result), {
+        initialDelay: 100,
+        maxAttempts: 5,
```
Comment (leite08) [inline packages/core/src/external/carequality/ihe-gateway-v2/ihe-gateway-v2-logic.ts:161]: Trying to minimize too many, too concentrated retries to the OSS API

File: packages/core/src/external/carequality/ihe-gateway/outbound-result-poller-lambda.ts [hunk 3]
```diff
@@ -28,67 +28,80 @@ export class OutboundResultPollerLambda extends OutboundResultPoller {
     this.docRetrievalLambdaName = docRetrievalLambdaName;
   }
 
-  isPDEnabled(): boolean {
-    return (this.patientDiscoveryLambdaName ?? "").trim().length > 0;
+  isPDEnabled(
+    lambdaName: string | undefined = this.patientDiscoveryLambdaName
+  ): lambdaName is string {
+    return (lambdaName ?? "").trim().length > 0;
   }
```
Comment (leite08) [inline packages/core/src/external/carequality/ihe-gateway/outbound-result-poller-lambda.ts:35]: This allows us to code w/ the lambda name below w/o having to have the non-null-assertion

File: packages/lambdas/src/sqs-to-converter.ts
Comment (leite08) [inline packages/lambdas/src/sqs-to-converter.ts:1]: To be fixed on https://github.com/metriport/metriport/pull/2137

File: packages/lambdas/src/sqs-to-fhir.ts
Comment (leite08) [inline packages/lambdas/src/sqs-to-fhir.ts:1]: To be fixed on https://github.com/metriport/metriport/pull/2137

**Review by jonahkaye — APPROVED** [review]
lgtm

**Pull request merged**